cdap-app-fabric/src/main/java/co/cask/cdap/app/preview/PreviewApplicationManager.java

```diff
@@ -27,8 +46,48 @@
  * @param <O> Output type.
  */
 public class PreviewApplicationManager<I, O> implements Manager<I, O> {
+
+  private final PipelineFactory pipelineFactory;
+  private final CConfiguration cConf;
+  private final Store store;
+  private final DatasetFramework datasetFramework;
+  private final DatasetFramework inMemoryDatasetFramework;
+  private final UsageRegistry usageRegistry;
+  private final ArtifactRepository artifactRepository;
+  private final Impersonator impersonator;
+  private final PrivilegesManager privilegesManager;
+  private final AuthenticationContext authenticationContext;
+
+  @Inject
+  PreviewApplicationManager(CConfiguration configuration, PipelineFactory pipelineFactory,
+                            Store store, DatasetFramework datasetFramework,
+                            @Named("datasetMDS") DatasetFramework inMemoryDatasetFramework,
+                            UsageRegistry usageRegistry, ArtifactRepository artifactRepository,
+                            PrivilegesManager privilegesManager,
+                            AuthenticationContext authenticationContext, Impersonator impersonator) {
+    this.cConf = configuration;
+    this.pipelineFactory = pipelineFactory;
+    this.store = store;
+    this.datasetFramework = datasetFramework;
+    this.inMemoryDatasetFramework = inMemoryDatasetFramework;
+    this.usageRegistry = usageRegistry;
+    this.artifactRepository = artifactRepository;
+    this.impersonator = impersonator;
+    this.privilegesManager = privilegesManager;
+    this.authenticationContext = authenticationContext;
+  }
+
   @Override
   public ListenableFuture<O> deploy(I input) throws Exception {
```

Comment:
This is called from https://github.com/caskdata/cdap/blob/develop/cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java#L582